File: dataset/Vanilla/OPENAI_gpt-5-2025-08-07/Task34.java
import java.util.*;

public class Task34 {
    public static boolean narcissistic(long n) {
        if (n <= 0) return false; // per problem, inputs are positive non-zero
        int digits = countDigits(n);
        long sum = 0;
        long temp = n;
        while (temp > 0) {
            int d = (int)(temp % 10);
            sum += ipow(d, digits);
            temp /= 10;
        }
        return sum == n;
    }

    private static int countDigits(long n) {
        int c = 0;
        while (n > 0) {
            c++;
            n /= 10;
        }
        return c;
    }

    private static long ipow(int base, int exp) {
        long result = 1;
        long b = base;
        int e = exp;
        while (e > 0) {
            if ((e & 1) == 1) result *= b;
            b *= b;
            e >>= 1;
        }
        return result;
    }

    public static void main(String[] args) {
        long[] tests = {153, 1652, 7, 370, 9474};
        for (long t : tests) {
            System.out.println(t + " -> " + narcissistic(t));
        }
    }
}

File: dataset/Vanilla - Copy/OPENAI_gpt-5-2025-08-07/Task34.cpp
#include <iostream>

long long ipow(long long base, int exp) {
    long long result = 1;
    long long b = base;
    int e = exp;
    while (e > 0) {
        if (e & 1) result *= b;
        b *= b;
        e >>= 1;
    }
    return result;
}

bool narcissistic(long long n) {
    if (n <= 0) return false;
    int digits = 0;
    long long temp = n;
    while (temp > 0) {
        digits++;
        temp /= 10;
    }
    long long sum = 0;
    temp = n;
    while (temp > 0) {
        int d = static_cast<int>(temp % 10);
        sum += ipow(d, digits);
        temp /= 10;
    }
    return sum == n;
}

int main() {
    std::cout << std::boolalpha;
    long long tests[] = {153, 1652, 7, 370, 9474};
    for (long long t : tests) {
        std::cout << t << " -> " << narcissistic(t) << "\n";
    }
    return 0;
}